packages/new-editor/src/components/editor/variables/detail/Value.tsx

```diff
@@ -0,0 +1,50 @@
+import { Fieldset, Input, SimpleSelect } from '@axonivy/ui-components';
+import { treeNodeValueAttribute } from '../../../../types/config';
+import { asEnumMetadata, metadataTypeDaytime, metadataTypeEnum, type Variable, type VariableUpdates } from '../../data/Variable';
+
+type ValueFieldsetProps = {
+  variable: Variable;
+  onChange: (updates: VariableUpdates) => void;
+};
+
+export const Value = ({ variable, onChange }: ValueFieldsetProps) => {
+  const filterEnumValuesForSelect = () => {
```

Comment:
why this name? IMO there is no filter functionality here.
This function creates the select items.

In this case I would use a `useMemo` hook. to make sure that this happens only once.

```ts 
const enumValues = useMemo(() => {
  const metadata = asEnumMetadata(variable.metadata);
  if (!metadata) {
    return [];
  }
  return Array.from(values)
    .filter(value => value !== '')
    .map(value => ({label: value, value}));
}, [variable])
```